File: JavaScript Fundamentals/Mid Exams/Mid Exam 3/movingTarget.js
function solve(arr) {
    let sequence = arr.shift().split(' ').map(x => Number(x));
    for (let el of arr) {
        if (el == 'End') break;
        let tokens = el.split(' ');
        if (tokens[0] == 'Shoot') {
            let index = Number(tokens[1]);
            let power = Number(tokens[2]);
            if (index >= 0 && index < sequence.length) {
                sequence[index] -= power;
                if (sequence[index] <= 0) sequence.splice(index, 1);
            }
        } else if (tokens[0] == 'Add') {
            let index = Number(tokens[1]);
            let value = Number(tokens[2]);
            if (index >= 0 && index < sequence.length) {
                sequence.splice(index, 0, value);
            } else console.log("Invalid placement!");
        } else {
            let index = Number(tokens[1]);
            let radius = Number(tokens[2]);
            let shotIndex = index - radius;
            let shots = (radius * 2) + 1;

            for (let i = 1; i <= shots; i++) {
                if (shotIndex >= 0 && shotIndex < sequence.length && sequence.length > 0) {
                    sequence.splice(shotIndex, 1);
                } else {
                    console.log("Strike missed!");
                    break;
                }
            }
        }
    }
    console.log(sequence.join('|'));
}

File: Python Fundamentals/Mid Exam problems/moving_target.py
targets = [int(x) for x in input().split()]

while True:
    command = input()

    if command == 'End':
        break

    command_list = command.split()
    action = command_list[0]
    index = int(command_list[1])

    if action == 'Shoot':
        if 0 <= index <= len(targets) - 1:
            power = int(command_list[2])
            targets[index] -= power

            if targets[index] <= 0:
                targets.pop(index)

    elif action == 'Add':
        if 0 <= index <= len(targets) - 1:
            value = int(command_list[2])
            targets.insert(index, value)
        else:
            print("Invalid placement!")

    elif action == 'Strike':
        radius = int(command_list[2])
        if 0 <= index <= len(targets) - 1:
            start_index = index - radius
            end_index = index + radius + 1

            if 0 <= start_index <= len(targets) - 1 and 0 <= end_index <= len(targets) - 1:
                for i in range(start_index, end_index):
                    targets.pop(start_index)
            else:
                print("Strike missed!")

result = [str(x) for x in targets]
print('|'.join(result))
